src/overlays/actors/ovl_Bg_Dblue_Movebg/z_bg_dblue_movebg.c

```diff
@@ -28,46 +59,829 @@ const ActorInit Bg_Dblue_Movebg_InitVars = {
+                }
+                break;
+
+            case 15:
+                if (!Flags_GetSwitch(globalCtx, arg1) ||
+                    (Flags_GetSwitch(globalCtx, arg1 + 1) && Flags_GetSwitch(globalCtx, arg1 + 2) &&
+                     Flags_GetSwitch(globalCtx, arg1 + 3))) {
+                    sp2C = 1;
+                }
+                break;
+        }
+    }
+    return sp2C;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Dblue_Movebg/func_80A29A80.s")
+#endif
+
+void BgDblueMovebg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgDblueMovebg* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80A2B978);
+    this->unk_160 = BGDBLUEMOVEBG_GET_F(thisx);
+    this->unk_1BC = BGDBLUEMOVEBG_GET_F000(thisx);
+    this->unk_1C0 = BGDBLUEMOVEBG_GET_FF0(thisx);
+    DynaPolyActor_Init(&this->dyna, 1);
+
+    if ((this->unk_160 == 9) || (this->unk_160 == 8)) {
+        if (D_80A2BBF4.unk_00 != 0) {
+            Actor_MarkForDeath(&this->dyna.actor);
+            return;
+        }
+        this->unk_170 = this->dyna.actor.room;
+        this->dyna.actor.room = -1;
+        D_80A2BBF4.unk_00 = 1;
+        this->unk_171 = this->dyna.actor.world.rot.z;
+        this->dyna.actor.world.rot.z = 0;
+        this->dyna.actor.shape.rot.z = 0;
+    } else if (this->unk_160 == 6) {
+        if (D_80A2BBF4.unk_01 != 0) {
+            Actor_MarkForDeath(&this->dyna.actor);
+            return;
+        }
+        this->unk_171 = this->dyna.actor.world.rot.z;
+        this->unk_170 = this->dyna.actor.room;
+        this->dyna.actor.world.rot.z = 0;
+        this->dyna.actor.shape.rot.z = 0;
+        this->dyna.actor.room = -1;
+        D_80A2BBF4.unk_01 = 1;
+    }
+
+    if (D_80A2B90C[this->unk_160] != NULL) {
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, D_80A2B90C[this->unk_160]);
+    }
+
+    this->unk_164 = D_80A2B8AC[this->unk_160];
+    this->unk_168 = D_80A2B8DC[this->unk_160];
+    this->unk_16C = D_80A2B93C[this->unk_160];
+
+    func_8013E3B8(&this->dyna.actor, this->unk_1B6, ARRAY_COUNT(this->unk_1B6));
+
+    switch (this->unk_160) {
+        case 1:
+            this->unk_1C4 = this->dyna.actor.world.rot.x;
+            this->unk_1C8 = this->dyna.actor.world.rot.z;
+            this->dyna.actor.world.rot.x = 0;
+            this->dyna.actor.shape.rot.x = 0;
+            this->dyna.actor.world.rot.z = 0;
+            this->dyna.actor.shape.rot.z = 0;
+            if (func_80A29A80(globalCtx, this->unk_1C8, this->unk_1C4)) {
+                this->unk_172 |= 1;
+            } else {
+                this->unk_172 &= ~1;
+            }
+            this->unk_17E = 0;
+            this->unk_184 = 0.0f;
+            if (Flags_GetSwitch(globalCtx, this->unk_1C0)) {
+                this->unk_18C = 0x384;
+            } else {
+                this->unk_18C = 0;
+            }
+            this->dyna.actor.shape.rot.y += (s16)((this->unk_18C / 10.0f) * 182.04445f);
```

Comment:
As long as it matches
```suggestion
            this->dyna.actor.shape.rot.y += (s16)((this->unk_18C / 10.0f) * (0x10000 / 360.0f));
```